src/main/java/org/reaktivity/nukleus/socks/internal/metadata/State.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright 2016-2017 The Reaktivity Project
+ *
+ * The Reaktivity Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.reaktivity.nukleus.socks.internal.metadata;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+/**
+    Marker annotation for methods that override MessageConsumer.accept but are called
```

Comment:
Is this check needed/required?

Reply:
Should provide a quick way to see what methods can become the current state held in a StreamProcessor's `private MessageConsumer streamState;`. They should be different from Throttling message consumers implementations for instance. For the comment about the signature check, I got confused, Java does it.